cocos/network/CCDownloader-curl.cpp

```diff
@@ -845,7 +847,7 @@ namespace cocos2d { namespace network {
             DLLOG("    DownloaderCURL: finish Task: Id(%d)", coTask.serialId);
         }
         
-        if (_impl->stoped())
+        if (!_breakScheduleFunc && _impl->stoped())
```

Comment:
I use this way to process class with sub-thread inside it on pc before. It works.
On mobile platform need tester do more test on variable device to confirm .